### ISSUE TITLE
`az ssh config` fails if provided with an ed25519 keypair

### ISSUE THREAD

**Issue by roy-work** [Azure/azure-cli-extensions#3520]
### Extension name (the extension in question)

`az ssh`

### Description of issue (in as much detail as possible)

If provided with an ed25519 keypair, `az ssh` fails to generate a config:
```
$ az ssh config -g my-resource-group -n my-vm -i ~/.ssh/id_ed25519 -p ~/.ssh/id_ed25519.pub --file ./config
Could not parse public key. Error: Public key is not ssh-rsa algorithm (ssh-ed25519)
```

Support for ed25519 would be nice. (I know Azure VMs _also_ do not support anything except RSA. There, too, support for ed25519 would be nice.)

**Comment by yonzhan** [comment]
SSH

**Comment by N6UDP** [comment]
I don't believe we have any current plans around non-RSA support but I agree it would be ideal to support other key types.

**Comment by N6UDP** [comment]
To be clear support needs to be added on the AAD service side first.
